script.js

```diff
@@ -47,3 +105,37 @@ function getIframeSrc(htmlContent) {
   const match = regex.exec(htmlContent);
   return match ? match[1] : '';
 }
+
+async function loadFromStorage() {
+  onSnapshot(collection(database, "quizzes"), (snapshot) => {
+    snapshot.forEach((doc) => {
+      const data = doc.data();
+      if (!data) return;
+      const quiz = stories.quizzes.find(q => q.id === data.quizId);
+      if (!quiz) return;
+      quiz.complete = data.complete;
+      quiz.score = data.score;
+    });
+    renderQuizLinks(document.getElementById("quizzes"), stories.quizzes);
```

Comment:
`stories.quizzes` is a global variable. Ye shall bring none of that global mutation heresy here!

Reply:
Bah! This be JarvarScript land! Everything be mutable!

Reply:
What changes would you suggest?

Reply:
I'd prefer `loadFromStorage` to return the quiz list so you can assign to the `stories` varaiable & the `renderQuizLinks` is a separate call under it because `loadFromStorage` doesn't really imply that it'd also `renderQuizLinks`